File: data/imported/knowledge/slides/net/merged/snippets/snippet_035.cs
[Fact]
    public void TestSolidFill()
    {
        // Solid fill colour persists after save/reload.
        using var pres = new Presentation();
        var slide = ClearSlide(pres);
        var shape = slide.Shapes!.AddAutoShape(ShapeType.Rectangle, 50, 50, 200, 100);
        shape.FillFormat.FillType = FillType.Solid;
        shape.FillFormat.SolidFillColor.Color = Color.FromArgb(255, 0, 128, 255);

        using var pres2 = TestHelpers.SaveAndReopen(pres, _tempDir);
        var ff = pres2.Slides[0].Shapes![0].FillFormat;
        ff.FillType.Should().Be(FillType.Solid);
        var c = ff.SolidFillColor.Color;
        c!.R.Should().Be(0);
        c.G.Should().Be(128);
        c.B.Should().Be(255);
    }

File: data/imported/knowledge/slides/cpp/merged/snippets/snippet_035.cpp
TEST_F(LineFormatIntegrationTest, MultipleDashStyles) {
    std::vector<Aspose::Slides::Foss::LineDashStyle> styles = {
        Aspose::Slides::Foss::LineDashStyle::SOLID,
        Aspose::Slides::Foss::LineDashStyle::DASH,
        Aspose::Slides::Foss::LineDashStyle::DOT,
        Aspose::Slides::Foss::LineDashStyle::DASH_DOT,
    };
    Presentation pres;
    auto& slide = pres.slides()[0];
    for (auto style : styles) {
        auto& shape = slide.shapes().add_auto_shape(ShapeType::RECTANGLE, 50, 50, 200, 50);
        shape.line_format().set_dash_style(style);
        EXPECT_EQ(shape.line_format().dash_style(), style);
    }
}